pkg/listener/listener.go

```diff
@@ -0,0 +1,132 @@
+		} else {
+			pods, _ = staticClient.CoreV1().Pods(pod.Namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: string(labels.Set(replicaSet.Spec.Selector.MatchLabels).AsSelector().String())})
+			podControllerName = replicaSet.GetName()
+		}
+	case "DaemonSet":
+		daemonSet, _ := staticClient.AppsV1().DaemonSets(pod.Namespace).Get(context.TODO(), pod.OwnerReferences[0].Name, metav1.GetOptions{})
+		pods, _ = staticClient.CoreV1().Pods(pod.Namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: string(labels.Set(daemonSet.Spec.Selector.MatchLabels).AsSelector().String())})
+		podControllerName = daemonSet.GetName()
+	case "StatefulSet":
+		statefulSet, _ := staticClient.AppsV1().StatefulSets(pod.Namespace).Get(context.TODO(), pod.OwnerReferences[0].Name, metav1.GetOptions{})
+		pods, _ = staticClient.CoreV1().Pods(pod.Namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: string(labels.Set(statefulSet.Spec.Selector.MatchLabels).AsSelector().String())})
+		podControllerName = statefulSet.GetName()
+	}
+
+	var containers []collector.ContainerProfile
+
+	// Merge all the container information of all the pods
+	for _, pod := range pods.Items {
+		typedObj, _ := dynamicClient.Resource(collector.AppProfileGvr).Namespace(pod.GetNamespace()).Get(context.TODO(), pod.GetName(), metav1.GetOptions{})
+		podApplicationProfileObj := getApplicationProfileFromObj(typedObj)
+
+		containers = append(containers, podApplicationProfileObj.Spec.Containers...)
+	}
+
+	// Fetch ApplicationProfile of the controller
+	_, err = dynamicClient.Resource(collector.AppProfileGvr).Namespace(pod.Namespace).Get(context.TODO(), podControllerName, metav1.GetOptions{})
+	if err != nil { // ApplicationProfile of controller doesn't exist so create a new one
+		controllerApplicationProfile := &collector.ApplicationProfile{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       collector.ApplicationProfileKind,
+				APIVersion: collector.ApplicationProfileApiVersion,
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: podControllerName,
+			},
+			Spec: collector.ApplicationProfileSpec{
+				Containers: containers,
+			},
+		}
+		controllerApplicationProfileRaw, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(controllerApplicationProfile)
+		_, _ = dynamicClient.Resource(collector.AppProfileGvr).Namespace(pod.Namespace).Create(context.TODO(), &unstructured.Unstructured{Object: controllerApplicationProfileRaw}, metav1.CreateOptions{})
+	} else { // ApplicationProfile of controller exists so update it
+		controllerApplicationProfile := &collector.ApplicationProfile{}
+		controllerApplicationProfile.Spec.Containers = containers
+		controllerApplicationProfileRaw, _ := json.Marshal(controllerApplicationProfile)
+		_, _ = dynamicClient.Resource(collector.AppProfileGvr).Namespace(pod.Namespace).Patch(context.TODO(), podControllerName, apitypes.MergePatchType, controllerApplicationProfileRaw, metav1.PatchOptions{})
+	}
+}
```

Comment:
What is unclear to me from the code is how the data is aggregated toward the owner reference.

I will write up a short proposal